# Exploring Ebay Car Sale Data

In this project, we will work with a dataset of used car from eBay Kleinanzeigen. It is a classifieds section of the German eBay website. This data was originally scraped and uploaded to kaggle. A few modifications are made to the original dataset
* 50,000 data points are sampled from the full dataset.

### The data's structure is described as follows:

| Column  | Description  | 
|---|---|
|  dateCrawled | When this ad was first crawled. All field-values are taken from this date  |
| name  | Name of the car  |
| seller  | Whether the seller is private or a dealer.  |
| offerType  | The type of listing  |
| price | The price on the ad to sell the car.  |
| abtest  | Whether the listing is included in an A/B test.  |
| vehicleType  | The vehicle Type.  |
| yearOfRegistration  | The year in which the car was first registered  |
| gearbox  | The transmission type  |
| powerPS  | The power of the car in PS  |
| model  |  The car model name.  |
| kilometer  | How many kilometers the car has driven  |
| monthOfRegistration  | The month in which the car was first registered  |
| fuelType  | What type of fuel the car uses  |
| brand  | The brand of the car  |
| notRepairedDamage  | If the car has a damage which is not yet repaired  |
| dateCreated  | The date on which the eBay listing was created  |
| nrOfPictures  | The number of pictures in the ad  |
| postalCode  | The postal code for the location of the vehicle  |
| lastSeenOnline  | When the crawler saw this ad last online  |


### Aim:
The aim of this project is to clean the data and analyze the included used car listings.

In [1]:
import numpy as np
import pandas as pd

In [2]:
autos = pd.read_csv('autos.csv', encoding= 'Latin-1')

Let's inspect the dataset:

In [3]:
print(autos.shape)
print(autos.info())
autos.head()

(50000, 20)
<class 'pandas.core.frame.DataFrame'>
RangeIndex: 50000 entries, 0 to 49999
Data columns (total 20 columns):
 #   Column               Non-Null Count  Dtype 
---  ------               --------------  ----- 
 0   dateCrawled          50000 non-null  object
 1   name                 50000 non-null  object
 2   seller               50000 non-null  object
 3   offerType            50000 non-null  object
 4   price                50000 non-null  object
 5   abtest               50000 non-null  object
 6   vehicleType          44905 non-null  object
 7   yearOfRegistration   50000 non-null  int64 
 8   gearbox              47320 non-null  object
 9   powerPS              50000 non-null  int64 
 10  model                47242 non-null  object
 11  odometer             50000 non-null  object
 12  monthOfRegistration  50000 non-null  int64 
 13  fuelType             45518 non-null  object
 14  brand                50000 non-null  object
 15  notRepairedDamage    40171 non-null  obje

dateCrawled                                               name  \
0  2016-03-26 17:47:46                   Peugeot_807_160_NAVTECH_ON_BOARD   
1  2016-04-04 13:38:56         BMW_740i_4_4_Liter_HAMANN_UMBAU_Mega_Optik   
2  2016-03-26 18:57:24                         Volkswagen_Golf_1.6_United   
3  2016-03-12 16:58:10  Smart_smart_fortwo_coupe_softouch/F1/Klima/Pan...   
4  2016-04-01 14:38:50  Ford_Focus_1_6_Benzin_TÜV_neu_ist_sehr_gepfleg...   

   seller offerType   price   abtest vehicleType  yearOfRegistration  \
0  privat   Angebot  $5,000  control         bus                2004   
1  privat   Angebot  $8,500  control   limousine                1997   
2  privat   Angebot  $8,990     test   limousine                2009   
3  privat   Angebot  $4,350  control  kleinwagen                2007   
4  privat   Angebot  $1,350     test       kombi                2003   

     gearbox  powerPS   model   odometer  monthOfRegistration fuelType  \
0    manuell      158  andere  150,000km                    3      lpg   
1  automatik      286     7er  150,000km                    6   benzin   
2    manuell      102    golf   70,000km                    7   benzin   
3  automatik       71  fortwo   70,000km                    6   benzin   
4    manuell        0   focus  150,000km                    7   benzin   

        brand notRepairedDamage          dateCreated  nrOfPictures  \
0     peugeot              nein  2016-03-26 00:00:00             0   
1         bmw              nein  2016-04-04 00:00:00             0   
2  volkswagen              nein  2016-03-26 00:00:00             0   
3       smart              nein  2016-03-12 00:00:00             0   
4        ford              nein  2016-04-01 00:00:00             0   

   postalCode             lastSeen  
0       79588  2016-04-06 06:45:54  
1       71034  2016-04-06 14:45:08  
2       35394  2016-04-06 20:15:37  
3       33729  2016-03-15 03:16:28  
4       39218  2016-04-01 14:38:50

This dataset consists of 50,000 rows and 20 columns. 15 columns are having a datatype of strings and 5 columns have 'int64' value.

Some columns have null values but non of them have more than 20% null value. 

### Convert camelCase to snakecase for column names:

This dataset also use 'camelCase' instead of python preferred 'Snakecase' for the column name. We have to make a conversion before the data analysis.m

In [4]:
print(autos.columns)
columns = autos.columns  

Index(['dateCrawled', 'name', 'seller', 'offerType', 'price', 'abtest',
       'vehicleType', 'yearOfRegistration', 'gearbox', 'powerPS', 'model',
       'odometer', 'monthOfRegistration', 'fuelType', 'brand',
       'notRepairedDamage', 'dateCreated', 'nrOfPictures', 'postalCode',
       'lastSeen'],
      dtype='object')


In [5]:
mapping_dict = {
    'dateCrawled':'date_crawled',
    'name':'name',
    'seller':'seller',
    'offerType':'offer_type',
    'price':'price',
    'abtest':'abtest',
    'vehicleType':'vehicle_type',
    'yearOfRegistration':'registration_year',
    'gearbox':'gearbox',
    'powerPS':'power_ps',
    'model':'model',
    'odometer':'odometer',
    'monthOfRegistration':'registration_month',
    'fuelType':'fuel_type',
    'brand':'brand',
    'notRepairedDamage':'unrepaired_damage',
    'dateCreated':'ad_created',
    'nrOfPictures':'nr_of_picture',
    'postalCode':'postal_code',
    'lastSeen':'last_seen'
}
autos.columns = autos.columns.map(mapping_dict)

In [6]:
autos.head()

date_crawled                                               name  \
0  2016-03-26 17:47:46                   Peugeot_807_160_NAVTECH_ON_BOARD   
1  2016-04-04 13:38:56         BMW_740i_4_4_Liter_HAMANN_UMBAU_Mega_Optik   
2  2016-03-26 18:57:24                         Volkswagen_Golf_1.6_United   
3  2016-03-12 16:58:10  Smart_smart_fortwo_coupe_softouch/F1/Klima/Pan...   
4  2016-04-01 14:38:50  Ford_Focus_1_6_Benzin_TÜV_neu_ist_sehr_gepfleg...   

   seller offer_type   price   abtest vehicle_type  registration_year  \
0  privat    Angebot  $5,000  control          bus               2004   
1  privat    Angebot  $8,500  control    limousine               1997   
2  privat    Angebot  $8,990     test    limousine               2009   
3  privat    Angebot  $4,350  control   kleinwagen               2007   
4  privat    Angebot  $1,350     test        kombi               2003   

     gearbox  power_ps   model   odometer  registration_month fuel_type  \
0    manuell       158  andere  150,000km                   3       lpg   
1  automatik       286     7er  150,000km                   6    benzin   
2    manuell       102    golf   70,000km                   7    benzin   
3  automatik        71  fortwo   70,000km                   6    benzin   
4    manuell         0   focus  150,000km                   7    benzin   

        brand unrepaired_damage           ad_created  nr_of_picture  \
0     peugeot              nein  2016-03-26 00:00:00              0   
1         bmw              nein  2016-04-04 00:00:00              0   
2  volkswagen              nein  2016-03-26 00:00:00              0   
3       smart              nein  2016-03-12 00:00:00              0   
4        ford              nein  2016-04-01 00:00:00              0   

   postal_code            last_seen  
0        79588  2016-04-06 06:45:54  
1        71034  2016-04-06 14:45:08  
2        35394  2016-04-06 20:15:37  
3        33729  2016-03-15 03:16:28  
4        39218  2016-04-01 14:38:50

To match with the python's preferred naming, we have modified the name of each column from camelcase to snakecase.

### Basic Data Exploration:
We will first look for:
* Text columns where all or almost all values are the same. These columns do not contain much information.
* Numerical data stored as text can be cleaned and converted.

#### To gather the descriptive statistics for all categorical and numerical columns

In [7]:
autos.describe(include='all')

date_crawled         name  seller offer_type  price abtest  \
count                 50000        50000   50000      50000  50000  50000   
unique                48213        38754       2          2   2357      2   
top     2016-03-29 23:42:13  Ford_Fiesta  privat    Angebot     $0   test   
freq                      3           78   49999      49999   1421  25756   
mean                    NaN          NaN     NaN        NaN    NaN    NaN   
std                     NaN          NaN     NaN        NaN    NaN    NaN   
min                     NaN          NaN     NaN        NaN    NaN    NaN   
25%                     NaN          NaN     NaN        NaN    NaN    NaN   
50%                     NaN          NaN     NaN        NaN    NaN    NaN   
75%                     NaN          NaN     NaN        NaN    NaN    NaN   
max                     NaN          NaN     NaN        NaN    NaN    NaN   

       vehicle_type  registration_year  gearbox      power_ps  model  \
count         44905       50000.000000    47320  50000.000000  47242   
unique            8                NaN        2           NaN    245   
top       limousine                NaN  manuell           NaN   golf   
freq          12859                NaN    36993           NaN   4024   
mean            NaN        2005.073280      NaN    116.355920    NaN   
std             NaN         105.712813      NaN    209.216627    NaN   
min             NaN        1000.000000      NaN      0.000000    NaN   
25%             NaN        1999.000000      NaN     70.000000    NaN   
50%             NaN        2003.000000      NaN    105.000000    NaN   
75%             NaN        2008.000000      NaN    150.000000    NaN   
max             NaN        9999.000000      NaN  17700.000000    NaN   

         odometer  registration_month fuel_type       brand unrepaired_damage  \
count       50000        50000.000000     45518       50000             40171   
unique         13                 NaN         7          40                 2   
top     150,000km                 NaN    benzin  volkswagen              nein   
freq        32424                 NaN     30107       10687             35232   
mean          NaN            5.723360       NaN         NaN               NaN   
std           NaN            3.711984       NaN         NaN               NaN   
min           NaN            0.000000       NaN         NaN               NaN   
25%           NaN            3.000000       NaN         NaN               NaN   
50%           NaN            6.000000       NaN         NaN               NaN   
75%           NaN            9.000000       NaN         NaN               NaN   
max           NaN           12.000000       NaN         NaN               NaN   

                 ad_created  nr_of_picture   postal_code            last_seen  
count                 50000        50000.0  50000.000000                50000  
unique                   76            NaN           NaN                39481  
top     2016-04-03 00:00:00            NaN           NaN  2016-04-07 06:17:27  
freq                   1946            NaN           NaN                    8  
mean                    NaN            0.0  50813.627300                  NaN  
std                     NaN            0.0  25779.747957                  NaN  
min                     NaN            0.0   1067.000000                  NaN  
25%                     NaN            0.0  30451.000000                  NaN  
50%                     NaN            0.0  49577.000000                  NaN  
75%                     NaN            0.0  71540.000000                  NaN  
max                     NaN            0.0  99998.000000                  NaN

This table gives a general insight of how the dataset distributed in each column. In this table, we have to look for columns that have a small or only 2 'unique' values. If the values of the column are biased, we can concluded that this column is not informative. Hence, it will be excluded from our analysis. 

To further our data cleaning, we have to check for columns:

'seller', 'offer_type', 'price', 'abtest', 'gearbox', 'odometer', 'fuel_type' and 'unrepaired_damage'. 

In this session, we will inspect these columns one by one.

In [8]:
autos['seller'].value_counts()

privat        49999
gewerblich        1
Name: seller, dtype: int64

This 'seller' column is heavily biased. 

In [9]:
autos['offer_type'].value_counts()

Angebot    49999
Gesuch         1
Name: offer_type, dtype: int64

This 'offer_type' column is heavily biased.

In [10]:
autos['price'].value_counts()

$0          1421
$500         781
$1,500       734
$2,500       643
$1,000       639
            ... 
$4,222         1
$4,475         1
$116,000       1
$5,849         1
$20,450        1
Name: price, Length: 2357, dtype: int64

This 'price' column provides informative understanding about the price of each computer.

In [11]:
autos['abtest'].value_counts()

test       25756
control    24244
Name: abtest, dtype: int64

In [12]:
autos['gearbox'].value_counts()

manuell      36993
automatik    10327
Name: gearbox, dtype: int64

In [13]:
autos['odometer'].value_counts()

150,000km    32424
125,000km     5170
100,000km     2169
90,000km      1757
80,000km      1436
70,000km      1230
60,000km      1164
50,000km      1027
5,000km        967
40,000km       819
30,000km       789
20,000km       784
10,000km       264
Name: odometer, dtype: int64

In [14]:
autos['fuel_type'].value_counts()

benzin     30107
diesel     14567
lpg          691
cng           75
hybrid        37
andere        22
elektro       19
Name: fuel_type, dtype: int64

In [15]:
autos['unrepaired_damage'].value_counts()

nein    35232
ja       4939
Name: unrepaired_damage, dtype: int64

'abtest', 'gearbox', 'odometer' and 'fuel_type' columns are not biased. We can also keep 'unrepaired_damage' column.

In short, 'seller', 'offer_type' columns will be removed from our analysis.

#### Price and odometer columns are numeric values but stored as text. 
* Remove any non-numeric characters
* Convert the column to a numeric dtype
* Rename the column if necessary (odometer to odometer_km for information)

In [16]:
autos['odometer'] = autos['odometer'].str.replace(',','')
autos['odometer'] = autos['odometer'].str.replace('km','')
autos['odometer'] = autos['odometer'].astype(int)
autos.rename({'odometer':'odometer_km'}, inplace = True, axis = 1)

In [17]:
autos['price'] = autos['price'].str.replace('$','')
autos['price'] = autos['price'].str.replace(',','')
autos['price'] = autos['price'].astype(int)

#### Inspect odometer_km column

In [18]:
odometer_km = autos['odometer_km']
odometer_km_unique_num = odometer_km.unique().shape[0]
print("Odometer_km has {} unique values".format(odometer_km_unique_num))
print('\n')
odometer_km_desc = odometer_km.describe()
print('Description of odometer_km')
print(odometer_km_desc)
print('\n')

print('Check the variations of odometer_km')
odometer_value_counts = odometer_km.value_counts()
print(odometer_value_counts)

odometer_value_counts_sorted = odometer_value_counts.sort_index(ascending = False)




Odometer_km has 13 unique values


Description of odometer_km
count     50000.000000
mean     125732.700000
std       40042.211706
min        5000.000000
25%      125000.000000
50%      150000.000000
75%      150000.000000
max      150000.000000
Name: odometer_km, dtype: float64


Check the variations of odometer_km
150000    32424
125000     5170
100000     2169
90000      1757
80000      1436
70000      1230
60000      1164
50000      1027
5000        967
40000       819
30000       789
20000       784
10000       264
Name: odometer_km, dtype: int64


There are 13 unique values in the odometer_km columns, the mean value is 124,732. There is no observable outliers in this column. Majority of cars have a odometer of 150,000 km.

#### Inspect the price column

In [19]:
price = autos['price']
price_unique_num = price.unique().shape[0]
print("price has {} unique values".format(price_unique_num))
print('\n')
price_desc = price.describe()
print('Description of price')
print(price_desc)
print('\n')

print('Check the variations of price')
price_value_counts = price.value_counts()
price_value_counts_sorted = price_value_counts.sort_index(ascending = False)
print(price_value_counts_sorted)
print('\n')

A = autos[(autos['price'] > 1.100000e+02) & (autos['price'] < 100000)]
autos_clean = autos[autos["price"].between(1.100000e+02, 100000)]
print('Cleaned dataset has {0} rows, {1} columns'.format(autos_clean.shape[0],autos_clean.shape[1]))


price has 2357 unique values


Description of price
count    5.000000e+04
mean     9.840044e+03
std      4.811044e+05
min      0.000000e+00
25%      1.100000e+03
50%      2.950000e+03
75%      7.200000e+03
max      1.000000e+08
Name: price, dtype: float64


Check the variations of price
99999999       1
27322222       1
12345678       3
11111111       2
10000000       1
            ... 
5              2
3              1
2              3
1            156
0           1421
Name: price, Length: 2357, dtype: int64


Cleaned dataset has 48051 rows, 20 columns


As we can observed, the mean value of the price column is about 9840. However, we found that there are extreme values existed in the price data. (eg. value 9999999 or value 0)

To filter out the outliers, we decided to set a upper bound price and lower bound price. In our data, we only accept data with reasonable price (ranging from 110 to 100000).

After the cleaning, our dataset has 48,051 rows and 20 columns.

### Now, we take a look at 'date_crowled', 'last_seen' and 'ad_created' columns:

These 3 columns are stored in string format. We have to convert the data inro a numerical representation for quantative analysis. Let's take a look at a few examples:

In [20]:
autos_clean[['date_crawled','ad_created','last_seen']][0:5]

date_crawled           ad_created            last_seen
0  2016-03-26 17:47:46  2016-03-26 00:00:00  2016-04-06 06:45:54
1  2016-04-04 13:38:56  2016-04-04 00:00:00  2016-04-06 14:45:08
2  2016-03-26 18:57:24  2016-03-26 00:00:00  2016-04-06 20:15:37
3  2016-03-12 16:58:10  2016-03-12 00:00:00  2016-03-15 03:16:28
4  2016-04-01 14:38:50  2016-04-01 00:00:00  2016-04-01 14:38:50

Let extract the date range from these columns and run a quick analysis:`

In [21]:
# Run distribution with missing values:
autosclean_value = autos_clean['date_crawled'].str[:10].value_counts(normalize = True, dropna = False)
print(autosclean_value.sort_index)

<bound method Series.sort_index of 2016-04-03    0.038605
2016-03-20    0.037772
2016-03-21    0.037273
2016-03-12    0.036961
2016-03-14    0.036669
2016-04-04    0.036586
2016-03-07    0.035983
2016-04-02    0.035629
2016-03-28    0.034984
2016-03-19    0.034755
2016-03-15    0.034214
2016-03-29    0.034110
2016-03-30    0.033756
2016-04-01    0.033652
2016-03-08    0.033215
2016-03-09    0.033007
2016-03-22    0.032840
2016-03-11    0.032674
2016-03-10    0.032382
2016-03-23    0.032299
2016-03-26    0.032216
2016-03-31    0.031841
2016-03-17    0.031508
2016-03-25    0.031404
2016-03-27    0.031113
2016-03-24    0.029448
2016-03-16    0.029448
2016-03-05    0.025369
2016-03-13    0.015733
2016-03-06    0.014068
2016-04-05    0.013090
2016-03-18    0.012841
2016-04-06    0.003163
2016-04-07    0.001394
Name: date_crawled, dtype: float64>


In [22]:
autosclean_value = autos_clean['ad_created'].str[:10].value_counts(normalize = True, dropna = False)
print(autosclean_value.sort_index)


<bound method Series.sort_index of 2016-04-03    0.038855
2016-03-20    0.037856
2016-03-21    0.037502
2016-04-04    0.036940
2016-03-12    0.036794
                ...   
2016-02-11    0.000021
2016-01-03    0.000021
2016-01-16    0.000021
2016-01-29    0.000021
2016-02-07    0.000021
Name: ad_created, Length: 76, dtype: float64>


In [23]:
autosclean_value = autos_clean['last_seen'].str[:10].value_counts(normalize = True, dropna = False)
print(autosclean_value.sort_index)

<bound method Series.sort_index of 2016-04-06    0.221993
2016-04-07    0.132214
2016-04-05    0.125034
2016-03-17    0.028095
2016-04-03    0.025140
2016-04-02    0.024849
2016-03-30    0.024661
2016-04-04    0.024536
2016-03-31    0.023870
2016-03-12    0.023850
2016-04-01    0.022892
2016-03-29    0.022330
2016-03-22    0.021394
2016-03-28    0.020790
2016-03-20    0.020666
2016-03-21    0.020561
2016-03-24    0.019687
2016-03-25    0.019126
2016-03-23    0.018584
2016-03-26    0.016649
2016-03-16    0.016420
2016-03-15    0.015837
2016-03-19    0.015775
2016-03-27    0.015504
2016-03-14    0.012632
2016-03-11    0.012424
2016-03-10    0.010655
2016-03-09    0.009552
2016-03-13    0.008886
2016-03-08    0.007305
2016-03-18    0.007305
2016-03-07    0.005390
2016-03-06    0.004308
2016-03-05    0.001082
Name: last_seen, dtype: float64>


In [24]:
print('Description for registration year column')
Au = autos_clean['registration_year'].describe()
print(Au)

print('Distribution for registration year column')
Dist = autos_clean['registration_year'].value_counts(normalize = True)
print(Dist[:30])

Description for registration year column
count    48051.000000
mean      2004.749433
std         88.052996
min       1000.000000
25%       1999.000000
50%       2004.000000
75%       2008.000000
max       9999.000000
Name: registration_year, dtype: float64
Distribution for registration year column
2000    0.064057
2005    0.060540
1999    0.059770
2004    0.056169
2003    0.056045
2006    0.055524
2001    0.054650
2002    0.051508
1998    0.048282
2007    0.047262
2008    0.045930
2009    0.043287
1997    0.039687
2011    0.033631
2010    0.032944
2017    0.028761
1996    0.028033
2012    0.027159
2016    0.024765
1995    0.024474
2013    0.016566
2014    0.013735
1994    0.012820
2018    0.009740
1993    0.008657
2015    0.007825
1992    0.007575
1991    0.006972
1990    0.006826
1989    0.003559
Name: registration_year, dtype: float64


### For column 'registration_year', there are outliers. 
A car can not be registered: 
* before year 1000: cars were invented after year 1000
* After year 9999: This is our future
* After year 2016: This dataset is obtained in 2016

By quick research, car was invented in year 1886. So, any data with registration outside year 1886 and year 2016 are considered as outliers.

In [25]:
lower_bound = 1886
upper_bound = 2016
autos_clean_regist = autos_clean[autos_clean["registration_year"].between(lower_bound, upper_bound)]


In [26]:
print(autos_clean_regist.shape[0])
print('Distribution of registration year')
autos_clean_regist['registration_year'].value_counts(normalize = True)


46181
Distribution of registration year


2000    0.066651
2005    0.062991
1999    0.062190
2004    0.058444
2003    0.058314
          ...   
1939    0.000022
1943    0.000022
1953    0.000022
1951    0.000022
1952    0.000022
Name: registration_year, Length: 78, dtype: float64

After removing outliers, the registration years for car are within year 1952 to 2016. There are 46181 rows of data remaining.

### Exploring variations between different car brands:
First, let's understand how many unique car brands existed in our dataset. For our analysis, we will select the top 20 most frequent car brands to calculate for their corresponding average car price.

In [38]:
brand = autos_clean_regist['brand']
unique_brand = brand.unique()
brand_value = brand.value_counts()

print(brand_value)
brand_key = brand_value.index
brand_key = brand_key[:20]
    

volkswagen        9770
bmw               5102
opel              4947
mercedes_benz     4476
audi              4017
ford              3214
renault           2170
peugeot           1380
fiat              1179
seat               844
skoda              760
nissan             709
mazda              702
smart              658
citroen            649
toyota             593
hyundai            464
sonstige_autos     435
volvo              422
mini               407
mitsubishi         377
honda              363
kia                328
alfa_romeo         309
suzuki             272
chevrolet          262
porsche            253
chrysler           163
dacia              123
daihatsu           115
jeep               106
land_rover          98
subaru              98
saab                77
jaguar              71
daewoo              68
trabant             63
rover               61
lancia              49
lada                27
Name: brand, dtype: int64


The first 20 car brands on the above table will be selected for our analysis.

In [40]:
brand_top = {}
for b in brand_key:
    select_rows = autos_clean_regist[autos_clean_regist['brand'] == b]
    sort_rows = select_rows.sort_values("brand", ascending = False)
    sort_price = sort_rows['price'].mean()
    brand_top[b] = sort_price

In [41]:
brand_top

{'volkswagen': 5452.7916069600815,
 'bmw': 8252.84711877695,
 'opel': 3019.591873862947,
 'mercedes_benz': 8575.378016085791,
 'audi': 9348.730395817774,
 'ford': 3765.181705040448,
 'renault': 2509.320737327189,
 'peugeot': 3122.5963768115944,
 'fiat': 2855.4444444444443,
 'seat': 4443.688388625593,
 'skoda': 6417.9118421052635,
 'nissan': 4769.795486600846,
 'mazda': 4152.736467236467,
 'smart': 3596.40273556231,
 'citroen': 3807.653312788906,
 'toyota': 5167.091062394604,
 'hyundai': 5411.075431034483,
 'sonstige_autos': 10993.505747126437,
 'volvo': 5004.803317535545,
 'mini': 10665.346437346438}

Among the pool of Ebay cars, the average price of 'mini', 'sonstige_autos', 'audi', 'bmw' and 'mercedes_benz'cars on Ebay are the most expensive. They are in general considered as 'high-class' brand in our daily life. We can also observe that there are market for cars with mid-range price and low-range price. Brands like 'Skoda', 'toyota', 'volkswagen', 'hyundai' are having an average price of around 5000. 

This shows that Ebay users may come from families of different wealth conditions. In the top 20 most frequent car brands, we can observe that both experience-price, mid-range price and low-price all contributes to the Ebay market.

In [45]:
brand_key = brand_value.index
brand_key = brand_key[:6]

brand_top_price = {}
brand_top_mileage = {}
for b in brand_key:
    select_rows = autos_clean_regist[autos_clean_regist['brand'] == b]
    sort_rows = select_rows.sort_values("brand", ascending = False)
    
    sort_price = sort_rows['price'].mean()
    sort_mileage = sort_rows['odometer_km'].mean()
    
    brand_top_price[b] = sort_price
    brand_top_mileage[b] = sort_mileage

In [53]:
price_series = pd.Series(brand_top_price)
mileage_series = pd.Series(brand_top_mileage)

df = pd.DataFrame(price_series, columns = ["mean price"])
df['mean mileage'] = mileage_series
df

mean price   mean mileage
volkswagen     5452.791607  128783.520983
bmw            8252.847119  132778.322227
opel           3019.591874  129379.421872
mercedes_benz  8575.378016  131117.068811
audi           9348.730396  129291.760020
ford           3765.181705  124271.935283